x-pack/plugins/triggers_actions_ui/public/application/sections/alert_details/components/alerts_route.tsx

```diff
@@ -36,11 +36,31 @@ export const AlertsRoute: React.FunctionComponent<WithAlertSummaryProps> = ({
   } = useKibana().services;
 
   const [alertSummary, setAlertSummary] = useState<AlertSummary | null>(null);
+  const [numberOfExecutions, setNumberOfExecutions] = useState(60);
+  const ruleID = useRef<string | null>(null);
 
   useEffect(() => {
-    getAlertSummary(rule.id, loadAlertSummary, setAlertSummary, toasts);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [rule]);
+    if (ruleID.current !== rule.id) {
+      ruleID.current = rule.id;
+      getAlertSummary(
+        ruleID.current,
+        loadAlertSummary,
+        setAlertSummary,
+        toasts,
+        numberOfExecutions
+      );
+    }
+  }, [rule, ruleID, numberOfExecutions, loadAlertSummary, setAlertSummary, toasts]);
+
+  const onChangeDuration = useCallback(
+    (executions: number) => {
+      setNumberOfExecutions(executions);
+      if (ruleID.current !== null) {
```

Comment:
Do we expect this scenario to happen? What does it mean for this to be null?

Reply:
No we don't expect this scenario to happen, this is for Typescript. Though I realize we could do the same thing by passing `ruleID.current!` to `getAlertSummary`.